### DISCUSSION TITLE
Does user relationship work with firebase auth?

### DISCUSSION THREAD

**Discussion by ELepolt** [firebase/emberfire#609] [Q&A] [answered]
Does emberfire utilize Firebase's built in authentication, or do we need to keep our own database of users?

**Comment by charlesfries** [comment, marked as answer]
I always end up keeping my own collection of users, indexed by the userID provided by Firebase. Emberfire does include authentication functions, but you need to keep your own collection of users, especially if you need to save custom attributes on your users (as far as I know).

Model hook of my `authenticated` route:

```
async model() {
  let { uid: id } = this.session.data.authenticated.user;

  let user;
  try {
    user = await this.store.findRecord('user', id);
  } catch(err) {
    if (err.message === 'The adapter could not find the resource') {
      this.store.recordForId('user', id).unloadRecord();
      user = this.store.createRecord('user', { id });
    }
  }

  if (!user) throw 'User not identified.';

  return user.save();
}
````
